graph/mutators/action.js

```diff
@@ -33,12 +33,18 @@ const createAction = async ({user = {}, pubsub, loaders: {Comments}}, {item_id,
 
   if (item_type === 'USERS' && action_type === 'FLAG') {
 
-    // Set the user as pending if it was a user flag.
-    await UsersService.setStatus(item_id, 'PENDING');
+    // Set the user as pending if it was a user flag and user has no Admin, Staff or Moderation roles
+    let user = await UsersService.findById(item_id);
+    if(!user.roles){
+      await UsersService.setStatus(item_id, 'PENDING');
+    }
   }
 
   if (comment) {
-    pubsub.publish('commentFlagged', comment);
+    let user = await UsersService.findById(comment.author_id);
+    if(!user.roles){
+      pubsub.publish('commentFlagged', comment);
+    }
```

Comment:
These changes only removes the live updates for staff comments. Can you revert these changes on the comment part? We will create a new ticket to tackle this as it requires a different approach. The reported queue shows comments that are flagged and neither approved nor rejected. It's too expensive to query whether the user is a staff or not for each comment to populate the list, so our approach is to mark staff comments as approved as soon as they are commented (and edit should not change that).

Reply:
Thanks for your work.

Reply:
Sure I can do that and maybe take a look at that other ticket once it is created